src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -39,7 +52,10 @@
      * @return map {user email : user}
      */
     public Map<String, User> groupByEmail(final List<User> users) {
-        return null;
+        return users.stream()
+                .distinct()
+                .collect(Collectors.toMap(User::getEmail, user -> user));
```

Comment:
`user -> user` nice:)
You might also use `Function.identity()` for the same purpose (no need to change it now, just for you to know).